src/plugins/data/server/search/search_service.ts

```diff
@@ -41,15 +42,18 @@ interface StrategyMap {
 export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
   private searchStrategies: StrategyMap = {};
 
-  constructor(private initializerContext: PluginInitializerContext) {}
+  constructor(
+    private initializerContext: PluginInitializerContext,
+    private readonly logger: Logger
+  ) {}
 
   public setup(
     core: CoreSetup<object, DataPluginStart>,
     { usageCollection }: { usageCollection?: UsageCollectionSetup }
   ): ISearchSetup {
     this.registerSearchStrategy(
       ES_SEARCH_STRATEGY,
-      esSearchStrategyProvider(this.initializerContext.config.legacy.globalConfig$)
+      esSearchStrategyProvider(this.initializerContext.config.legacy.globalConfig$, this.logger)
```

Comment:
what if we pass only 1 argument (`this.initializerContext`) into `esSearchStrategyProvider` ?

Reply:
Same here, the logger is generated from the context, but I wouldn't want to be re-creating it everywhere.
No strong feelings here though.